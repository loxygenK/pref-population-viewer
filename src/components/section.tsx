import styles from "./section.module.scss";

export interface SectionProps {
  children: React.ReactNode;
  title: string;
  className?: string;
}
export const Section: React.FC<SectionProps> = ({
  children,
  title,
  className,
}) => {
  return (
    <section className={className}>
      <h2 className={styles.section_heading}>{title}</h2>
      {children}
    </section>
  );
};
