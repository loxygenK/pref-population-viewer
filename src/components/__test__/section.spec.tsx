import renderer from "react-test-renderer";
import { Section } from "../section";

describe("Section wrapper component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Section title="Section">
          <p>Random content</p>
        </Section>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
