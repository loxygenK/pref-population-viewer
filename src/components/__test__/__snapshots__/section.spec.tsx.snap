// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section wrapper component renders correctly 1`] = `
<section>
  <h2
    className="section_heading"
  >
    Section
  </h2>
  <p>
    Random content
  </p>
</section>
`;
